src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,104 @@
+        Scanner scanner = new Scanner(file);
+        //System.out.println("parsing activated");
+
+        while (scanner.hasNext()) {
+            //this is for formating
+            String[] task = scanner.nextLine().split("[\\[||\\]]");
+            String taskContentHelper = task[4].split(" ", 2)[1];//the task and time without the brackets
+            String taskContent = taskContentHelper.split("\\(")[0].trim();//just the task
+            //System.out.println("task main content" + taskContentHelper);
+            //System.out.println("task1 " + taskContent);
+            String taskMark = task[3];
+            String taskType = task[1];
+            Task newTask;
+            // System.out.println("the mark:" + taskMark);
+
+            if (taskType.equals("T")) {
+                //System.out.println("todo parsed");
+                newTask = new ToDo(taskContent);
+                if (taskMark.equals("X")) {
+                    newTask.setDone();
+                }
+                inputs.addTask(newTask);
+            } else if (taskType.equals("D")) {
+                //System.out.println("event parsed");
+                String[] timingContent = taskContentHelper.split("[\\(||\\)]");//this is to split up and obtain time segment
+                System.out.println("timingContent: " + timingContent[1]);
+                //small problem where the user may not specify porperly
+                String time[] = timingContent[1].split(" ");//array of the elements in the time
+                String dateHelper = "";// the specific date
+                for (int i = 1; i < time.length; i++) {
+                    dateHelper += time[i] + " ";
+                }
+                String date = dateHelper.trim();
+                //System.out.println(date + "no");
+                newTask = new Deadline(taskContent, date);
+                if (taskMark.equals("X")) {
+                    newTask.setDone();
+                }
+                inputs.addTask(newTask);
+            } else {
+                String[] timingContent = taskContentHelper.split("[\\(||\\)]");//this is to split up and obtain time segment
+                //System.out.println("timingContent: " + timingContent[1]);
+                //small problem where the user may not specify porperly
+                String time[] = timingContent[1].split(" ");//array of the elements in the time
+                String dateHelper = "";// the specific date
+                for (int i = 1; i < time.length; i++) {
+                    dateHelper += time[i] + " ";
+                }
+                String date = dateHelper.trim();
+                //System.out.println(date + "no");
+                newTask = new Event(taskContent, date);
+                if (taskMark.equals("X")) {
+                    newTask.setDone();
+                }
+                inputs.addTask(newTask);
+            }
+        }
+    }
```

Comment:
I really appreciate how you organize if-else blocks and for loop with good coding styles.